doc/faq.md

```diff
@@ -85,24 +85,28 @@ DPVS side TOA format is defined in `include/ipvs/proto_tcp.h`, while option code
   };
   ... ...
   struct tcpopt_addr {
-     uint8_t opcode;
-     uint8_t opsize;
-     uint16_t port;
-     uint32_t addr;
+      uint8_t opcode;
+      uint8_t opsize;
+      __be16 port;
+      uint8_t addr[16];
   } __attribute__((__packed__));
   ```
 
+> In case of nat64, RS's codes need a little changes to get real client IP/port. DPVS provide an example TCP server and a nginx patch for nat64 toa. Pls refer to [example_nat64](../kmod/toa/example_nat64).
+
 <a id="udp" />
 
-### Is DPVS support UDP ? How to get UDP Client Real IP/port ?
+### Does DPVS support UDP ? How to get UDP Client Real IP/port ?
+
+Yes, it do support UDP. In order to get real client IP/port in FullNAT mode, you need install UOA module on RS, and do a little code change. Uoa support two modes: `opp` for private protocol mode, which supports IPv4/IPv6/Nat64, and `ipo` for ip option mode, which support IPv4 only. Pls refer to [uoa.md](../kmod/uoa/uoa.md) and [udp_serv.c](../kmod/uoa/example/udp_serv.c).
 
-Yes, it do support UDP. In order to get real client IP/port in FullNAT mode, you need install UOA module on RS, and do a little code change. Pls refer `uoa/uoa.md` and `uoa/example/udp_serv.c`.
+> Which uoa mode should choose? Honestly speaking, choose the one that works, because unlike TCP option, either IP option or private IP protocol is often restricted by network devices or policies. Thus consider your network environments and choose the one best suitable.
```

Comment:
Which uoa mode should choose?  -> Which UOA mode should **I** choose? / Which UOA mode to choose?
and choose the **best suitable** one.